mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,509 @@
+
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
+        let payment_amount = try_tx_s!(wei_from_big_decimal(&args.amount, self.decimals));
+        let (maker_secret_hash, taker_secret_hash) = match args.tx_type_with_secret_hash {
+            SwapTxTypeWithSecretHash::MakerPaymentV2 {
+                maker_secret_hash,
+                taker_secret_hash,
+            } => (maker_secret_hash, taker_secret_hash),
+            _ => {
+                return Err(TransactionErr::Plain(ERRL!(
+                    "Unsupported swap tx type for timelock refund"
+                )))
+            },
+        };
+        let args = {
+            let taker_address = public_to_address(&Public::from_slice(args.taker_pub));
+            MakerRefundArgs {
+                payment_amount,
+                taker_address,
+                taker_secret: [0u8; 32],
+                taker_secret_hash: try_tx_s!(taker_secret_hash.try_into()),
+                maker_secret_hash: try_tx_s!(maker_secret_hash.try_into()),
+                payment_time_lock: args.time_lock,
+                token_address,
+            }
+        };
+        let data = try_tx_s!(self.prepare_refund_maker_payment_timelock_data(args).await);
+
+        self.sign_and_send_transaction(
+            U256::from(ZERO_VALUE),
+            Action::Call(maker_swap_v2_contract),
+            data,
+            U256::from(gas_limit),
+        )
+        .compat()
+        .await
+    }
+
+    pub(crate) async fn refund_maker_payment_v2_secret_impl(
+        &self,
+        args: RefundMakerPaymentSecretArgs<'_, Self>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let (token_address, gas_limit) = self
+            .gas_limit_v2
+            .gas_limit(
+                &self.coin_type,
+                EthPaymentType::MakerPayments,
+                PaymentMethod::RefundSecret,
+            )
+            .map_err(|e| TransactionErr::Plain(ERRL!("{}", e)))?;
+
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
+        let taker_secret = try_tx_s!(args.taker_secret.try_into());
+        let maker_secret_hash = try_tx_s!(args.maker_secret_hash.try_into());
+        let payment_amount = try_tx_s!(wei_from_big_decimal(&args.amount, self.decimals));
+        let args = {
+            let taker_address = public_to_address(args.taker_pub);
+            MakerRefundArgs {
+                payment_amount,
+                taker_address,
+                taker_secret,
+                taker_secret_hash: [0u8; 32],
```

Comment:
Same here

Reply:
fixed here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/72ed3060ace61d9a14eca6fc5cedeb957ad0be54

also did this in taker file